### PR TITLE
DOC: remove constrained_layout kwarg from examples

### PR DESCRIPTION
**Pull request by rcomer**
## PR Summary

The form `constrained_layout=True` is discouraged in the relevant [docstrings](https://matplotlib.org/devdocs/api/figure_api.html#matplotlib.figure.Figure) since v3.5 (#19892), but is still used across the examples.  This PR updates the examples with the preferred form.  For the strings, I tried to choose single vs double quotes to be consistent with any strings on nearby lines.  Also puts instances of the phrase "constrained layout" in italics, consistent with the decision for the constrained layout tutorial at #25144.

The tutorials and user guide also have `constrained_layout=True` in places, but I think that can be a separate PR.

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: examples/ticks/date_index_formatter.py [hunk 1]
```diff
@@ -18,19 +18,19 @@
...
 import matplotlib.lines as ml
 from matplotlib.ticker import Formatter
 
 # Load a numpy record array from yahoo csv data with fields date, open, high,
 # low, close, volume, adj_close from the mpl-data/sample_data directory. The
 # record array stores the date as an np.datetime64 with a day unit ('D') in
 # the date column (``r.date``).
-r = (cbook.get_sample_data('goog.npz', np_load=True)['price_data']
-     .view(np.recarray))
+r = cbook.get_sample_data('goog.npz', np_load=True)['price_data'].view(np.recarray)
```
Comment (rcomer) [inline examples/ticks/date_index_formatter.py:29]: I know we said we wouldn't re-flow, but this line was crying out for it!

File: examples/ticks/date_index_formatter.py [hunk 2]
```diff
@@ -18,19 +18,19 @@
...
 r = r[:9]  # get the first 9 days
 
 fig, (ax1, ax2) = plt.subplots(nrows=2, figsize=(6, 6),
-                               constrained_layout={'hspace': .15})
+                               layout=ConstrainedLayoutEngine(hspace=0.15))
```
Comment (jklymak) [inline examples/ticks/date_index_formatter.py:33]: I don't mind this, but obscure imports are sometimes not nice.  Maybe:

```suggestion
                               layout='constrained')
fig.get_layout_engine.set(hspace=0.15)
```

**Review by jklymak — APPROVED** [review]
One small nit.  Could self merge if you take it or not so long as the doc build and flake8 pass.

**Review by oscargus — APPROVED** [review]
As I am not updated on the 3.7 status, I am not merging, but looks good!

File: examples/ticks/date_index_formatter.py (hunk 2 above)
Comment (rcomer) [reply examples/ticks/date_index_formatter.py]: I don't feel strongly either way, so done.

**Comment by rcomer** [comment]
I’ll take a look at the backport after 3.7.0 is out.  Thanks for the reviews!

**Pull request merged**